### PR TITLE
[Alignment] Add operators; refactor to use `pytest`

### PR DESCRIPTION
**Pull request by awgu**
**Overview**
- Adds alignment checks for `layernorm`, `view` -> `embedding`, and `getitem` when used as a reshape
- Refactors the alignment infra to use `pytest` for the final comparison of FlexFlow vs. PyTorch tensors
- Adds initial utilities to bisect the full mT5 model for more fine-grained alignment

**Notes**
- Some operators do not fit well into the framework of "run the forward pass through the operator, compute the loss against some 32-bit dtype label, and run the backward pass".
    - For example, the `view` operator on an `int64` input yields an `int64` output. Without a subsequent operator to follow the `view`, there is no way to compute the loss and hence check that gradients propagate through the operator correctly. (This is why I checked `view` -> `embedding` together. `embedding` is the only way I know of to transform data from `int64` to `float32`.)
- HuggingFace's symbolic trace only allows tracing a selection of classes, so I cannot directly trace a `T5Stack` for example. The workaround is to always trace the full mT5 encoder-decoder model and extract from that trace in the Python FlexFlow space.
- Next, I plan to tackle the mT5 encoder only or maybe even a subproblem of that. Some challenges include:
    - Dropout introduces non-determinism that cannot be aligned. One solution is to introduce something like the PyTorch [`model.eval()`](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.eval) API to disable dropout in FlexFlow if that does not already exist.
    - FlexFlow's `layernorm` operator is aligned with PyTorch's `LayerNorm` (which is great!), but mT5 uses its own [`T5LayerNorm`](https://github.com/huggingface/transformers/blob/e0a515407518657e021b7f4547be83333b520932/src/transformers/models/t5/modeling_t5.py#L231), which omits the bias and does not subtract the mean. For convenience of alignment, we should support a version of FlexFlow's `layernorm` that mirrors `T5LayerNorm`.
    - I need to devise a reliable and programmatic way to batch export and batch import model parameter values from FlexFlow to PyTorch. Doing this for a model with a single operator is easy, but when doing it in batch, I have to ensure that the I can properly line up the exported/imported values (e.g. so that the first FlexFlow `linear`'s weight lines up with the PyTorch first `linear`'s weight, etc.).

### CODE REVIEW

**Review by jiazhihao — APPROVED** [review]

**Comment by awgu** [comment]
Note to self: Remember to squash the commits before merging.
